src/backend/joanie/payment/models.py

```diff
@@ -1 +1,465 @@
-# Create your models here.
+"""
+Declare and configure models for the payment part
+"""
+import uuid
+from decimal import Decimal as D
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+from django.dispatch import receiver
+from django.utils import timezone
+from django.utils.functional import cached_property
+from django.utils.translation import get_language
+from django.utils.translation import gettext_lazy as _
+
+from babel.numbers import get_currency_symbol
+from djmoney.models.fields import MoneyField
+from djmoney.money import Money
+from howard.issuers import InvoiceDocument
+from parler.utils import get_language_settings
+from parler.utils.context import switch_language
+
+from joanie.core.models import Order
+from joanie.core.utils import merge_dict
+
+from . import enums as payment_enums
+from . import get_payment_backend
+
+User = get_user_model()
+
+
+class Invoice(models.Model):
+    """
+    Invoice model is an accounting element related to an order
+    """
+
+    parent = models.ForeignKey(
+        "self",
+        on_delete=models.RESTRICT,
+        related_name="children",
+        verbose_name="parent",
+        blank=True,
+        null=True,
+    )
+    reference = models.CharField(
+        db_index=True,
+        editable=False,
+        max_length=40,
+        unique=True,
+    )
+    created_on = models.DateTimeField(
+        _("created on"),
+        auto_now_add=True,
+        editable=False,
+    )
+    updated_on = models.DateTimeField(
+        _("updated on"),
+        auto_now=True,
+        editable=False,
+    )
+    order = models.OneToOneField(
+        Order,
+        verbose_name=_("order"),
+        on_delete=models.PROTECT,
+        blank=True,
+        null=True,
```

Comment:
Is there a case of an invoice not linked to an order?

Reply:
As I resolved your [comment](https://github.com/openfun/joanie/pull/51#discussion_r777904044) below, each invoice is now linked to an order.